### PR TITLE
Feature/tech 8102

### PR DESCRIPTION
**Pull request by Sameer-Begh**
Campaign listing page API Integration

### CODE REVIEW

File: src/app/page.tsx [hunk 1]
```diff
@@ -3,24 +3,129 @@ import {Box, Grid} from '@mantine/core';
...
+  const featuredURL = apiEndpoints.getFeaturedCampaignList(
+    CAMPAIGN_PARAM_STATUS,
+    true,
+    10,
+    true,
   );
+  const {data} = useGetResponseQuery(URL);
+  const {data: featuredCampaign} = useGetResponseQuery(featuredURL);
+
+  console.log('useGetResponseQuery featuredList data >>>>', featuredCampaign); //reference
```
Comment (shivams10) [inline src/app/page.tsx:22]: remove console

File: src/app/page.tsx [hunk 2]
```diff
@@ -3,24 +3,129 @@ import {Box, Grid} from '@mantine/core';
...
+          title,
+          partnerName,
+          cardImgUrls,
+          nextUserPromptTxt,
+          endsAtTimestamp,
+          campaignStatus,
+          claimRewardType,
+          nextUserPrompt,
+          campaignID,
+        } = {} as any,
```
Comment (shivams10) [inline src/app/page.tsx:42]: define proper type remove any

File: src/app/page.tsx [hunk 3]
```diff
@@ -3,24 +3,129 @@ import {Box, Grid} from '@mantine/core';
...
+
+  const renderCampaignBanner = () => {
+    return featuredCampaignData?.map((campaign: any) => {
+      const {
+        campaignInfo: {
+          title,
+          bannerImgUrls,
+          endsAtTimestamp,
+          campaignStatus,
+        } = {} as any,
```
Comment (shivams10) [inline src/app/page.tsx:87]: define proper type remove any

File: src/app/profile/page.tsx [hunk 4]
```diff
@@ -1,37 +1,23 @@
...
   const account = useAccount();
   const {address, chain, isConnected} = account;
   const {signMessageAsync: signMessage} = useSignMessage();
 
-  const handleMetamaskConnect = () => {
-    open();
-  };
-
   const onSignSuccess = (data: string) => {
     console.log({onSuccessData: data});
```
Comment (shivams10) [inline src/app/profile/page.tsx:22]: remove console

File: src/app/profile/page.tsx [hunk 5]
```diff
@@ -52,41 +38,11 @@ const Profile: React.FC = () => {
       }
     };
     if (signature && isConnected) {
-      setIsMetaMaskConnected(true);
-      setWalletAddress(address);
       console.log({chain});
```
Comment (shivams10) [inline src/app/profile/page.tsx:41]: remove console

File: src/components/cards/campaignCard/index.tsx [hunk 6]
```diff
@@ -7,36 +7,45 @@ import './campaignCard.scss';
...
+  const {color, bgColor, strokeColor, showProgressBar} =
+    campaignButtonStyle[statusStyle] || ({} as ICampaignButtonStyle);
 
-  const _progressBarText = showProgressBar
-    ? `(${_completedTask}/${_totalTask})`
+  const progressBarText = showProgressBar
+    ? `(${completedTask}/${totalTask})`
     : '';
 
   const _progressBarWidth = showProgressBar
```
Comment (shivams10) [inline src/components/cards/campaignCard/index.tsx:37]: why are we using this naming convention?

File: package-lock.json
Comment (shivams10) [inline package-lock.json:1]: don't push package-lock.json file untill you are installing any new library.
Pls revert back

File: src/app/page.tsx (hunk 3 above)
Comment (mandeepatindigg) [reply src/app/page.tsx:94]: we will resolve this in next review, for now merging it

**Pull request merged**